<?xml version="1.0"?>
<launch>

<!-- <include file="$(find explore_large_map)/launch/large_map.launch"/> -->

<!-- common param for astar_planner_pkg -->
<arg name="local_map_frame_name" value="base_link"/>
<arg name="global_map_frame_name" value="/odom"/>
<arg name="abso_global_map_frame_name" value="/abso_odom"/>
<arg name="allow_time_transmission_delay_ms_" value="1000"/>

<!-- mkz
<arg name="robot_length" value="4.93"/>
<arg name="robot_width" value="2.45"/>
<arg name="robot_wheelbase" value="2.85"/>
<arg name="base2back" value="1.100"/>
<arg name="minimum_turning_radius" value="5.5"/>
-->

<!-- toyataã€€-->
<arg name="robot_length" value="4.9"/>
<arg name="robot_width" value="1.95"/>
<arg name="robot_wheelbase" value="2.86"/>
<arg name="base2back" value="1.0"/>
<arg name="minimum_turning_radius" value="5"/>


<!-- private param for explore_large_map_pkg : /global_map,/explore_entry_map ; /move_base_simple/goal, /local_search_goal /-->
<arg name="re_planner_map_topic_name" value="/global_map"/>
<arg name="re_goal_pose_topic_name" value="/local_search_goal"/>

    <!-- for test astar actionlib
    <node pkg="astar_planner" type="astar_action_client_node" name="astar_action_client_node" output="screen" />
    -->

    <node pkg="astar_planner" type="astar_action_server_node" name="astar_action_server_node" output="screen" >
        <param name="global_map_frame_name" value="$(arg global_map_frame_name)" />
        <param name="local_map_frame_name" value="$(arg local_map_frame_name)" />
        <param name="send_path_topic_name" value="/global_path" />

        <param name="path_plan_update_cycle_" value="0.08" />
        <param name="distance_threhold_to_goal_" value="2" />

    </node>

     <arg name="config" default="$(find sas_space_explore)/cfg/params.yaml" />
      <env name="ROSCONSOLE_CONFIG_FILE"
             value="$(find sas_space_explore)/cfg/custom_rosconsole.conf"/>
      <node pkg="sas_space_explore" type="get_plan_server_node" name="get_plan_server_node" output="screen">
        <rosparam command="load" file="$(arg config)"/>
        <param name="angle_size" value="32" />
        <param name="heading_size" value="32" />
        <param name="allow_use_last_path" value="true" />
        <param name="allow_offset_distance" value="2" />
      </node>

       <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sas_space_explore)/rviz/sas_explore_node_viz.rviz"/>
       <node pkg="rqt_reconfigure"  type="rqt_reconfigure" name="reconfigure_gui"/>
       <node pkg="dynamic_reconfigure" type="dynparam" name="dynparam" args="load /get_plan_server_node $(find sas_space_explore)/cfg/dynamic_params.yaml"/>



    <!--
    <node pkg="rviz" type="rviz" name="rviz_astar" args="-d $(find astar_planner)/rviz/astar_planner_service.rviz"/>
    -->
</launch>
